### ISSUE TITLE
"progress" formatter math seems wrong -- creates 0 width bars when there should be room for at least one pixel

### ISSUE THREAD

**Issue by hossman** [olifolkerd/tabulator#919]
Consider the following simple example:
```
    $(document).ready(function() {
      $("#example-table").tabulator({
        columns:[
          {title:"Name", field:"name", width:200},
          {title:"Age", field:"age", width:200, formatter:"progress"} ],
        data:[
          {id:1, name:"Five", age:5},
          {id:2, name:"Four", age:4}
        ]
      });
    });
```

In my testing with tabulator 3.4.1 and the tabulator.min.css (both via cdnjs.cloudflare.com) in a browser window plenty wide for both columns to have the width=200, the Cell for the first "age" progress bar (in the "Five" row) looks like this...

```
<div title="" aria-label="95" tabulator-field="age" style="width: 200px; min-width: 30px; position: relative; height: 25px;" role="gridcell" class="tabulator-cell">
  <div style="position:absolute; top:8px; bottom:8px; left:4px; right:95%; margin-right:4px; background-color:#2DC214; display:inline-block;" data-max="100" data-min="0"></div>
  <div class="tabulator-col-resize-handle"></div>
  <div class="tabulator-col-resize-handle prev"></div>
</div>
```
...resulting in the inner div (with the `background-color:#2DC214` having a computed width of "1.95" pixels.

Meanwhile, for the "Four" row, this is what's produced for the "age" cell...

```
<div title="" aria-label="96" tabulator-field="age" style="width: 200px; min-width: 30px; position: relative; height: 25px;" role="gridcell" class="tabulator-cell">
  <div style="position:absolute; top:8px; bottom:8px; left:4px; right:96%; margin-right:4px; background-color:#2DC214; display:inline-block;" data-max="100" data-min="0"></div>
  <div class="tabulator-col-resize-handle"></div>
  <div class="tabulator-col-resize-handle prev"></div>
</div>
```

...resulting in the inner div (with the `background-color:#2DC214` having a computed width of "0" pixels.

I gather what's happening here is that value of the cell is being used to compute a "percentage" (relative the min/max which default to 0/100) which is then subtracted from the 100% to produce the "right" style attribute -- but these calculations don't seem to be taking into account the `left:4px;` and `margin-right:4px;` and how those change what exactly that "96%" is going to be relative to.

NOTE: I'm not a css/DOM expert, so forgive me if i'm being naive -- but it seems like if the cell is 200px wide, then even if there is a 4px "padding" on both sides of the "max theoretical bar length", then the overall "bar length" for an effective field value of 4% should be at least 1px ... even if the effective cell value was only 1% shouldn't the bar length be at least 1px? ... `(200-4-4)/100 = 1.92` right?

**Comment by olifolkerd** [comment]
Hey @hossman 

Thanks for your detailed message, you are indeed correct, there is a small mathematical bug in the progress bar calculations, i will have a look into that this evening and include a fix in the 3.4.4 patch release this weekend.

Cheers

Oli :)

**Comment by olifolkerd** [comment]
Hey @hossman 

I have just pushed a fix for this to the master branch and will include it in this weekends patch release.

I also fixed another issue with the progress formatter where it wasn't respecting the columns ***align*** property.

Thanks for letting me know about that

Oli :)
